Userland/DynamicLoader/main.cpp

```diff
@@ -44,20 +46,73 @@ static void perform_self_relocations(auxv_t* auxvp)
         exit(1);
 }
 
-static void display_help()
+static ErrorOr<int> open_executable(StringView path)
 {
-    char const message[] =
-        R"(You have invoked `Loader.so'. This is the helper program for programs that
-use shared libraries. Special directives embedded in executables tell the
-kernel to load this program.
-
-This helper program loads the shared libraries needed by the program,
-prepares the program to run, and runs it. You do not need to invoke
-this helper program directly. If you still want to run a program with the loader,
-run: /usr/lib/Loader.so [ELF_BINARY])";
-    fprintf(stderr, "%s", message);
+    int rc = open(path.characters_without_null_termination(), O_RDONLY | O_EXEC, 0666);
+    if (rc < 0)
+        return Error::from_errno(errno);
+    int checked_fd = rc;
+    ArmedScopeGuard close_on_failure([checked_fd] {
+        close(checked_fd);
+    });
+
+    struct stat executable_stat;
+    rc = fstat(checked_fd, &executable_stat);
+    if (rc < 0)
+        return Error::from_errno(errno);
+    if (!S_ISREG(executable_stat.st_mode)) {
+        if (S_ISDIR(executable_stat.st_mode))
+            return Error::from_errno(EISDIR);
+        return Error::from_errno(EINVAL);
+    }
+
+    close_on_failure.disarm();
+    return checked_fd;
+}
+
+// FIXME: Find a better way to supply the environment variables to serenity_main.
+char** g_env;
+
+ErrorOr<int> serenity_main(Main::Arguments arguments)
```

Comment:
I personally think the changes to use LibMain and ArgsParser here are overkill. 

There are two options: the command, and whether we want secure mode.  The Dynamic loader startup should be simple and straightforward. We can easily parse those manually without adding a dependency on LibCore to the Dynamic Loader. It already uses too much code, dragging LibCore into it is too much I think. As is LibMain.

Reply:
I added other options to the application, so it's not only `--secure` now. Manually parsing all of them is a no-go imho.

Reply:
I'm fine with `ArgsParser`, but I don't understand why we need to use LibMain in DynamicLoader. We are not even `TRY()`ing anything (good!).